Index: code/base/common/src/com/tc/util/VendorVmSignature.java
===================================================================
--- code/base/common/src/com/tc/util/VendorVmSignature.java	(revision 15913)
+++ code/base/common/src/com/tc/util/VendorVmSignature.java	(working copy)
@@ -15,6 +15,7 @@
 
   private static final String OS_WINDOWS          = "win32";
   private static final String OS_LINUX            = "linux";
+  private static final String OS_FREEBSD          = "freebsd";
   private static final String OS_SOLARIS_SPARC    = "solaris";
   private static final String OS_MAC_OSX          = "osx";
   private static final String OS_SOLARIS_X86      = "solaris-x86";
@@ -97,6 +98,7 @@
 
     if (lowerCaseOS.startsWith("windows")) { return OS_WINDOWS; }
     if (lowerCaseOS.startsWith("linux")) { return OS_LINUX; }
+    if (lowerCaseOS.startsWith("freebsd")) { return OS_FREEBSD; }
     if (lowerCaseOS.startsWith("aix")) { return OS_AIX; }
     if (lowerCaseOS.startsWith("mac") || lowerCaseOS.startsWith("darwin")) { return OS_MAC_OSX; }
     if (lowerCaseOS.startsWith("sunos")) {
