Index: src/main/java/net/sf/ehcache/Cache.java
===================================================================
--- src/main/java/net/sf/ehcache/Cache.java	(revision 2243)
+++ src/main/java/net/sf/ehcache/Cache.java	(working copy)
@@ -1971,7 +1971,9 @@
      * @throws IllegalStateException if the cache is already {@link Status#STATUS_SHUTDOWN}
      */
     public synchronized void dispose() throws IllegalStateException {
-        checkStatusNotDisposed();
+        if (checkStatusAlreadyDisposed()) {
+            return;
+        }
 
         if (executorService != null) {
             executorService.shutdown();
@@ -2163,11 +2165,9 @@
             throw new IllegalStateException("The " + configuration.getName() + " Cache is not alive.");
         }
     }
-
-    private void checkStatusNotDisposed() throws IllegalStateException {
-        if (status.equals(Status.STATUS_SHUTDOWN)) {
-            throw new IllegalStateException("The " + configuration.getName() + " Cache is disposed.");
-        }
+    
+    private boolean checkStatusAlreadyDisposed() throws IllegalStateException {
+        return status.equals(Status.STATUS_SHUTDOWN);
     }
 
 
