Index: jgroupsreplication/src/main/java/net/sf/ehcache/distribution/jgroups/JGroupsCacheManagerPeerProviderFactory.java
===================================================================
--- jgroupsreplication/src/main/java/net/sf/ehcache/distribution/jgroups/JGroupsCacheManagerPeerProviderFactory.java	(revision 10208)
+++ jgroupsreplication/src/main/java/net/sf/ehcache/distribution/jgroups/JGroupsCacheManagerPeerProviderFactory.java	(working copy)
@@ -20,7 +20,6 @@
 import net.sf.ehcache.CacheManager;
 import net.sf.ehcache.distribution.CacheManagerPeerProvider;
 import net.sf.ehcache.distribution.CacheManagerPeerProviderFactory;
-import net.sf.ehcache.util.ClassLoaderUtil;
 import net.sf.ehcache.util.PropertyUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -58,7 +57,7 @@
                 LOG.warn("Both '" + CONNECT + "' and '" + FILE + "' properties set. '" + CONNECT + "' will be ignored");
             }
             
-            final ClassLoader contextClassLoader = ClassLoaderUtil.getStandardClassLoader();
+            final ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
             final URL configUrl = contextClassLoader.getResource(file);
             
             LOG.debug("Creating JGroups CacheManagerPeerProvider for {} with configuration file: {}", cacheManager.getName(), configUrl);
